### PR TITLE
fix(components): [input] missing autofocus types

### LINKED ISSUE #14243: [TypeScript] [input] autocomplete causes error with vueCompilerOptions.strictTemplates enabled
Bug Type: **`TypeScript`**

## Environment

- Vue Version: `3.3.4`
- Element Plus Version: `2.3.12`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36`
- Build Tool: `Nuxt`

### Related Component

- `el-input`

### Reproduction Link

[Link](https://stackblitz.com/edit/nuxt-starter-pcggtq?file=tsconfig.json,app.vue)

### Steps to reproduce

Just watch the console or execute `npm run typecheck`.

### What is Expected?

No error.

### What is actually happening?

Error on the `autofocus` property of `<el-input>`.
```
app.vue:3:28 - error TS2345: Argument of type '{ modelValue: string; autofocus: boolean; }' is not assignable to parameter of type '{ readonly type?: string | undefined; readonly modelValue?: EpPropMergeType<(new (...args: any[]) => string | number) | (() => string | number | null | undefined) | ((new (...args: any[]) => string | number) | (() => string | number | null | undefined))[], unknown, unknown>; ... 44 more ...; onClear?: (() => any) | ...'.
  Object literal may only specify known properties, and 'autofocus' does not exist in type '{ readonly type?: string | undefined; readonly modelValue?: EpPropMergeType<(new (...args: any[]) => string | number) | (() => string | number | null | undefined) | ((new (...args: any[]) => string | number) | (() => string | number | null | undefined))[], unknown, unknown>; ... 44 more ...; onClear?: (() => any) | ...'.

3     <ElInput v-model="msg" autofocus />
                             ~~~~~~~~~


Found 1 error in app.vue:3
```

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by HeftyKoo**
closed #14243

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### <samp>🤖 Generated by Copilot at 25aecd1</samp>

This pull request adds a new `autofocus` prop to the `ElInput` component, which enables the native autofocus attribute for the input or textarea element. This prop can be used to automatically focus the input element when the component is rendered. The changes affect the files `input.vue` and `input.ts` in the `packages/components/input/src` directory.

## Related Issue

Fixes #\_\_\_.

### <samp>🤖 Generated by Copilot at 25aecd1</samp>

*  Add `autofocus` prop to `inputProps` interface to support native autofocus attribute for `ElInput` component ([link](https://github.com/element-plus/element-plus/pull/14246/files?diff=unified&w=0#diff-5dba1faa003ddc1780d877ed72854dfa7bb7e10e574631516d3bc85622242ed9R169-R174))
*  Bind `autofocus` prop to `autofocus` attribute of native input element in `input.vue` template for single input and textarea cases ([link](https://github.com/element-plus/element-plus/pull/14246/files?diff=unified&w=0#diff-740318b89e9e14e0ecf7296023cfe04687fea7371612db4f9b5d456d1648b2a7R45), [link](https://github.com/element-plus/element-plus/pull/14246/files?diff=unified&w=0#diff-740318b89e9e14e0ecf7296023cfe04687fea7371612db4f9b5d456d1648b2a7R122))

### CODE REVIEW

File: packages/components/input/src/input.ts [hunk 1]
```diff
@@ -166,6 +166,13 @@ export const inputProps = buildProps({
     type: definePropType<StyleValue>([Object, Array, String]),
     default: () => mutable({} as const),
   },
+  /**
+   * @description native input autofocus
+   */
+  autofocus: {
+    type: Boolean,
+    default: false,
```
Comment (btea) [inline packages/components/input/src/input.ts:174]: The default value here may be omitted.

Comment (HeftyKoo) [reply packages/components/input/src/input.ts:174]: I saw other boolean props have default value, so i just want them keep same.

**Comment by btea** [comment]
The input already has attrs bound to it, I'm not sure if adding an autofocus attribute is necessary. 🤔
https://github.com/element-plus/element-plus/blob/d791229a6aaf695f451b6c4afedc2fd7fa764b79/packages/components/input/src/input.vue#L33

**Comment by HeftyKoo** [comment]
> The input already has attrs bound to it, I'm not sure if adding an autofocus attribute is necessary. 🤔
> 
> https://github.com/element-plus/element-plus/blob/d791229a6aaf695f451b6c4afedc2fd7fa764b79/packages/components/input/src/input.vue#L33

Yes, it has attrs bound to it, but adding it to the prop is for the type-safe as issue #14243 describes.

**Review by btea — APPROVED** [review]

**Pull request merged**